test/test_mahalanobis_mixin.py

```diff
@@ -11,7 +11,12 @@
 from sklearn.datasets import make_spd_matrix, make_blobs
 from sklearn.utils import check_random_state, shuffle
 from sklearn.utils.multiclass import type_of_target
-from sklearn.utils.testing import set_random_state
+import sklearn
+from packaging import version
+if version.parse(sklearn.__version__) >= version.parse('0.22.0'):
```

Comment:
This seems like something we could do once during module initialization, then have a global constant like `SKLEARN_AT_LEAST_0_22` that we could use.

Or make a `sklearn_shims.py` file that does the conditional import, and have everything else import from that.

Reply:
I agree, will do

Reply:
done